src/scale/Log.ts

```diff
@@ -86,8 +86,9 @@ class LogScale extends Scale {
 
     setExtent(start: number, end: number): void {
         const base = this.base;
-        start = mathLog(start) / mathLog(base);
-        end = mathLog(end) / mathLog(base);
+        // log(-Infinity) is NaN, so safe guard here
+        start = mathLog(Math.max(0, start)) / mathLog(base);
+        end = mathLog(Math.max(0, end)) / mathLog(base);
```

Comment:
The result of `mathLog(base)` can be cached and shared via tweaking the first line of code.

Reply:
If cache is required, maybe it's better to be implemented inside `mathLog`. But first of all, do we need to cache this? Is there any performance problem issued? If not, making a cache takes more space and may not be worthy the price.

Reply:
I just means we can simply tweak the first line to be
```js
const base = mathLog(base)
```
It is calculated twice. I think it is unnecessary.

Reply:
Oh. Thanks! I misunderstood.